irods/rule.py

```diff
@@ -65,8 +66,15 @@ def load(self, rule_file, encoding = 'utf-8'):
         self.body = '@external\n'
 
         # parse rule file
-        with io_open(rule_file, encoding = encoding) as f:
+        with (io_open(rule_file, encoding = encoding) if isinstance(rule_file,six.string_types) else rule_file
```

Comment:
consider the same here, make it plain what is happening and why - this is logic, the (missing) part that caused the bug.

Reply:
ok , did this by adding a doc string.

Reply:
the final commit will contain a brief summary of the same info

Reply:
gotcha.